cli/cli.go

```diff
@@ -0,0 +1,246 @@
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+var preferences *prefs.Preferences
+
+var argMap = arg.Map{
+	ARG_NO_COLOR: {Type: arg.BOOL},
+	ARG_HELP:     {Type: arg.BOOL, Alias: "u:usage"},
+	ARG_VER:      {Type: arg.BOOL, Alias: "ver"},
+}
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func Init() {
+	args, errs := arg.Parse(argMap)
+
+	if len(errs) != 0 {
+		fmtc.Println("Arguments parsing errors:")
+
+		for _, err := range errs {
+			fmtc.Printf("  %s\n", err.Error())
+		}
+
+		os.Exit(1)
+	}
+
+	if arg.GetB(ARG_NO_COLOR) {
+		fmtc.DisableColors = true
+	}
+
+	if arg.GetB(ARG_VER) {
+		showAbout()
+		return
+	}
+
+	if arg.GetB(ARG_HELP) || len(args) == 0 {
+		showUsage()
+		return
+	}
+
+	switch len(args) {
+	case 0:
+		showUsage()
+		return
+	case 1:
+		processCommand(args[0], nil)
+	default:
+		processCommand(args[0], args[1:])
+	}
+}
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func prepare() {
+	preferences = prefs.New(CONFIG_FILE)
+	errs := preferences.Validate()
+
+	if len(errs) > 0 {
+		for _, err := range errs {
+			terminal.PrintErrorMessage(err.Error())
+		}
+		os.Exit(1)
+	}
+}
+
+// process start source processing
+func processCommand(cmd string, args []string) {
+	var err error
+
+	prepare()
+
+	switch cmd {
+	case CMD_ADD, CMD_ADD_SHORTCUT:
+		err = addCommand(args)
+	case CMD_DELETE, CMD_DELETE_SHORTCUT:
+		err = deleteCommand(args)
+	case CMD_LIST, CMD_LIST_SHORTCUT:
+		err = listCommand()
+	case CMD_INFO, CMD_INFO_SHORTCUT:
+		err = infoCommand(args)
+	case CMD_HELP:
+		showUsage()
+	default:
+		fmtc.Printf("{r}Unknown command{!}\n")
+		os.Exit(ERROR_UNSUPPORTED)
+	}
+
+	if err != nil {
+		fmtc.Printf(err.Error())
+		os.Exit(ERROR_UNSUPPORTED)
+	}
+}
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func addCommand(args []string) error {
+	var err error
+
+	if len(args) != 3 {
+		err = fmtc.Errorf("Unable to handle %v arguments\n", len(args))
+		return err
+	}
+
+	var (
+		//src = args[0]
+		name = args[1]
+		//version = args[2]
+	)
+
+	r := repo.New(preferences, name)
+
+	r.AddBox("qweqwe")
+
+	return err
+}
+
+func deleteCommand(args []string) error {
+	var err error
+
+	if len(args) != 1 {
+		err = fmtc.Errorf("Unable to handle %v arguments\n", len(args))
+	} else {
+		var name = args[0]
+
+		fmtc.Println(name)
+	}
+
+	return err
+}
+
+func listCommand() error {
+	return nil
+}
+
+func infoCommand(args []string) error {
+	var err error
+
+	if len(args) != 1 {
+		err = fmtc.Errorf("Unable to handle %v arguments\n", len(args))
+	} else {
+		var name = args[0]
+		fmtc.Println(name)
+	}
+
+	return err
+}
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func showUsage() {
+	info := usage.NewInfo(APP)
+
+	info.AddCommand(CMD_ADD, "Add image to the Vagrant repository")
```

Comment:
Add command arguments to command info like this:
```
info.AddCommand(CMD_ADD, "Add image to the Vagrant repository", "name", "version")
```

ek.usage also support optional command arguments. If you want to mark arguments as optional, just add `?` prefix to name, like this:
```
info.AddCommand(CMD_ADD, "Add image to the Vagrant repository", "?name", "?version")
```